GU- Blood Brain Analysis Part II
========================================================


```{r setup, echo=FALSE}
opts_chunk$set(tidy=TRUE, echo=TRUE, highlight=TRUE, figalign='center', fig.height=9, fig.width=9, out.width='800px', message=FALSE, error=TRUE, warning=FALSE, cache=FALSE)

# Setup report details
clientname="Ben Andreone"
clientemail="bandreon4119@gmail.com"
lablocation="Neurobiology at HMS"
analystname="Meeta Mistry"
analystemail="mmistry@hsph.harvard.edu"

```

Array analysis for `r clientname` (`r clientemail`) at `r lablocation`. Contact `r analystname` (`r analystemail`) for additional details. Request from client was:
  
> Compare the gene expression levels to those from other endothelial cell microarray data sets published in GEO. Two specific datasets of interest: 1) GSE9202 This study compared expression in E18.5 and adult mouse brain endothelial cells to samples from the rest of the brain 2) GSE35802 This study compared expression in endothelial cells between brain and lung at E14.5, P7.5, and adult. They used a different platform (Agilent 4x44K), but it may be interesting to qualitatively compare their data to our timepoint.

# Bioconductor and R libraries used
```{r, libraries, echo=TRUE}

library(affy)
library(arrayQualityMetrics)
library(RColorBrewer)
library(simpleaffy)
library(limma)
library(pheatmap)
library(xtable)
library(plyr)
library(ggplot2)
library(gplots)
library(GEOquery)
library(simpleaffy)


# Grab John Hutchinson's convenience functions
source("http://dl.dropboxusercontent.com/u/4253254/Resources/functions.r")
source("~/R/scripts/useful functions/roc.R")
```

### Get variables
- get base directory for analyses
- specify data and results directories
- specify column headers used in metadata file


```{r variables, echo=TRUE}
# Setup directory variables
baseDir <- '.'
dataDir <- file.path(baseDir, "data")
metaDir <- file.path(dataDir, "meta")
resultsDir <- file.path(baseDir, "results")
```

### Load the expression data

```{r dataimport GEO, echo=TRUE}

# Load GEO data

# Affymetrix GeneChip Mouse Genome 430 2.0 microarrays. Array data was processed using the Affy and gcrma packages
gse1 <- getGEO('GSE9202', destdir=file.path(dataDir, 'geo'))

# Agilent 4x44K 
gse2 <- getGEO('GSE35802', destdir=file.path(dataDir, 'geo'))

eset1 <- as(gse1$GSE9202_series_matrix.txt.gz, "ExpressionSet")
eset2 <- as(gse2$GSE35802_series_matrix.txt.gz, "ExpressionSet")

# Gu Dataset
gu.data <- read.affy('covars.desc', path='data/Gu/', verbose=F)
eset3 <- call.exprs(gu.data, "rma")
fData(eset3) <- fData(eset1)

```

```{r metadata extraction, eval=FALSE, echo=FALSE}
# Extract metadata eset1
title <- as.character(pData(eset1)$title)
v <- vapply(title, function(x)as.character(strsplit(x, ",", fixed=TRUE)[[1]]),
        c("Specimen" = "", "Timept" = "", "Reps" = ""))
colnames(v) <- rownames(pData(eset1))
timept <- as.vector(sapply(timept, function(x) strsplit(x, ": ", fixed=T)[[1]][2]))
pheno1 <- data.frame(t(v))
write.table(pheno1, file=file.path(metaDir, 'GSE9202.pheno.txt'), sep="\t", quote=F)

# Extract metadata eset2
title <- as.character(pData(eset2)$title)
Timept <- as.character(pData(eset2)$characteristics_ch1.1)
Timept <- as.vector(sapply(Timept, function(x) strsplit(x, ": ", fixed=T)[[1]][2]))
Specimen  <- as.character(pData(eset2)$characteristics_ch1.2)
Specimen <- as.vector(sapply(Specimen, function(x) strsplit(x, ": ", fixed=T)[[1]][2]))
Reps<-rep(1:5, 6)
# One adult liver/lung replicate was not used as it failed QC
Reps<-Reps[-length(Reps)]

pheno2<-data.frame(cbind(title, timept, specimen, reps))
rownames(pheno2) <- rownames(pData(eset2))
write.table(pheno2, file=file.path(metaDir, 'GSE35802.pheno.txt'), sep="\t", quote=F)
               
```


### Load the metadata: GSE9202

```{r import metadata1, results='asis'}

meta1 <- read.table(file.path(metaDir, 'GSE9202.pheno.txt'), header=TRUE, sep='\t', na.strings='NULL', row.names=1)
pData(eset1) <- meta1

# Table of attributes
kable(table(meta1$Timept, meta1$Specimen), format='html')
mosaicplot(table(meta1$Timept, meta1$Specimen), main='Categorial variables')

```


### Load the metadata: GSE35802
```{r import metadata2, results='asis'}

meta2 <- read.table(file.path(metaDir, 'GSE35802.pheno.txt'), header=TRUE, sep='\t', na.strings='NULL', row.names=1)
pData(eset2) <- meta2

# Table of attributes
kable(table(meta2$Timept, meta2$Specimen), format='html')
mosaicplot(table(meta2$Timept, meta2$Specimen), main='Categorial variables')

```

### Collapse all three datasets to gene-level for cross-platform comparison: FOR GENE-LEVEL COMPARISON (option 1)
```{r collapse, echo=FALSE, eval=FALSE}

source("~/R/scripts/useful functions/collpaseProbes.R")

collapse.geo1 <-collapseProbes(eset1, "Gene Symbol") 
collapse.geo2 <- collapseProbes(eset2, "GENE_SYMBOL")
collapse.gu <- collapseProbes(eset3, "Gene Symbol")

# Use only intersecting set of genes
igenes <- intersect(row.names(collapse.geo1), row.names(collapse.geo2))

# Keep only genes present in the intersection
collapse.geo1 <- collapse.geo1[which(rownames(collapse.geo1) %in% igenes), ]
collapse.geo2 <- collapse.geo2[which(rownames(collapse.geo2) %in% igenes), ]
collapse.gu <- collapse.gu[which(rownames(collapse.gu) %in% igenes), ]

# Reassign eset objects with collapsed data
datasets <- c("collapse.geo1", "collapse.geo2", "collapse.gu")

for (d in 1:length(datasets)){
  
  eset <- new("ExpressionSet", exprs=as.matrix(get(datasets[d])))
  pData(eset) <- pData(get(paste0("eset", d)))
  assign(paste0("eset", d), eset)
}

```

## GSE9202

> Array data was processed using the Affy and gcrma packages in the Bioconductor project. Student t test was used to evaluate differential expression. The false discovery rate method was used to perform multiple test correction for probability values.The log2 change was calculated as the log2 of the average expression difference between E18.5 microvessels and “rest-of-brain” tissue or adult brain microvessels.



### QC Analysis
ArrayQualityMetrics QC report for [GSE9202](./results/report_raw_GSE9202/index.html)


```{r QC_report1, echo=TRUE, eval=FALSE}
 arrayQualityMetrics(expressionset=eset1, intgroup=c('Specimen', 'Timept'),
                     outdir='./results/report_raw_GSE9202', force=TRUE,  do.logtransform=FALSE)
```
 

### Differential expression analysis
Following a similar [example](http://www.bioconductor.org/help/course-materials/2005/BioC2005/labs/lab01/estrogen/)

```{r GSE9202 analysis, results='asis'}

# Get contrasts
f <- paste(meta1$Specimen,meta1$Timept,sep="")
f <- factor(f)
design <- model.matrix(~0+f)
colnames(design) <- levels(f)


# Fit a linear model
fit <- lmFit(eset1, design)

# Define a comtrast model matrix
cont.matrix <- makeContrasts(byage="MicrovascularFragmentsE18.5-MicrovascularFragments1yearold",
                             bytissue="MicrovascularFragmentsE18.5-RestOftheBrainE18.5",levels=design)

# Extract the linear model fit for the contrasts
fit2  <- contrasts.fit(fit, cont.matrix)
fit2  <- eBayes(fit2)

# Set threshold 
p.cutoff <- 0.001
logfc.cutoff <- 2

# Get significant genes
gene_list<-topTable(fit2, coef=2, number=nrow(exprs(eset1)))
gene_list$threshold.FDR = as.factor(abs(gene_list$logFC) > logfc.cutoff 
                                    & gene_list$adj.P.Val < p.cutoff)
top.genes<-gene_list[which(as.logical(gene_list$threshold.FDR)),]

results <- decideTests(fit2, adjust.method="fdr", p.value=p.cutoff, lfc=logfc.cutoff)
colnames(results) <- c('Age Effects', 'Tissue Effects')

GSE9202 <- gene_list
vennDiagram(results)
```

### Cross-reference tissue effects against results from the Gu dataset re-analysis
The Venn diagram illustrates the number of overlapping signigicant probes between the two datasets. The ROC curve is used to illustrate where the Gu dataset significant probes appear with respect to the ranking of GSE9202 genes (ranked by adjusted p-value).

```{r cross-reference1_option1, echo=FALSE, eval=FALSE}

# FOR GENE-LEVEL COMPARISON eval=TRUE
mod<-model.matrix(~ -1 + Tissue + SampleNumber, pData(eset3))

# Fit a linear model
fit<-lmFit(eset3, mod)

# Compute estimated coefficients and standard errors for contrasts
contrasts <- makeContrasts(TissueCortex-TissueLung, levels=mod)
fit2<- contrasts.fit(fit, contrasts)
fit2<-eBayes(fit2)

# Set threshold 
p.cutoff <- 0.001
logfc.cutoff <- 2

# Get significant genes
gene_list<-topTable(fit2, coef=1, number=nrow(exprs(eset3)))
gene_list$threshold.FDR = as.factor(abs(gene_list$logFC) > logfc.cutoff 
                                    & gene_list$adj.P.Val < p.cutoff)
GU <- gene_list
gu.reanalysis <- GU[which(as.logical(GU$threshold.FDR)), ]
```

```{r cross-reference1_option2, echo=FALSE, results='asis'}

# Pull out top genes and find the overlap
g.geo <- rep(0, nrow(GSE9202))
g.geo[which(rownames(GSE9202) %in% rownames(top.genes))] <- 1
g.gu <- rep(0, nrow(GSE9202))

# If option1 is run, comment the following line
gu.reanalysis <- read.delim(file.path(dataDir, 'gu_reanalysis.txt'), header=T, sep="\t", row.names=1, as.is=T)
g.gu[which(rownames(GSE9202) %in% rownames(gu.reanalysis))] <- 1

overlap <- (cbind(as.logical(g.geo), as.logical(g.gu)))
rownames(overlap) <- rownames(GSE9202)
colnames(overlap) <- c("GSE9202", "Gu Dataset")

vennDiagram(overlap)

# Use ROC to see where the Gu genes lie within the ranking
ordered <- GSE9202[order(GSE9202$adj.P.Val), ]
roc <- rocRanked(rownames(gu.reanalysis), ordered)

```


## GSE35802

> GSE35802: A linear model was fit to the filtered data using the limma package from Bioconductor and comparisons
were analyzed between brain and liver/lung tissue at each developmental stage (embryo, pup, or adult). Criteria for DE was log 2 FC > 1 and adjusted p-value < 0.05 (Benjamini-Hochberg correction). As expected, microarray analysis identified several Wnt/beta-catenin target genes and other previously characterized BBB-marker genes i.e Glut1, P-gp, Meca32, Lef1


### QC Analysis
ArrayQualityMetrics QC report for [GSE35802](./results/report_raw_GSE35802/index.html)

```{r QC_report2, echo=TRUE, eval=FALSE}

 arrayQualityMetrics(expressionset=eset2, intgroup=c('Specimen', 'Timept'),
                     outdir='./results/report_raw_GSE35802', force=TRUE,  do.logtransform=FALSE)

```
 
 
### Differential expression analysis

```{r GSE35802 analysis, results='asis'}

# Get contrasts
f <- paste(meta2$Specimen,meta2$Timept,sep="")
f <- factor(f)
design <- model.matrix(~0+f)
colnames(design) <- levels(f)
colnames(design)[4:6]<-c("liverlungadult", "liverlungembryo", "liverlungpup")

# Remove NAs
exprs(eset2)<-na.omit(exprs(eset2))

# Fit a linear model
fit <- lmFit(eset2, design)

# Define a comtrast model matrix
cont.matrix <- makeContrasts(adult="brainadult-liverlungadult", pup="brainpup-liverlungpup",
                             embryonic="brainembryo-liverlungembryo",levels=design)


# Extract the linear model fit for the contrasts
fit2  <- contrasts.fit(fit, cont.matrix)
fit2  <- eBayes(fit2)

# Set threshold 
p.cutoff <- 0.001
logfc.cutoff <- 2

# Get significant genes
gene_list<-topTable(fit2, coef=3, number=nrow(exprs(eset2)))
gene_list$threshold.FDR = as.factor(abs(gene_list$logFC) > logfc.cutoff 
                                    & gene_list$adj.P.Val < p.cutoff)
top.genes<-gene_list[which(as.logical(gene_list$threshold.FDR)),]

results <- decideTests(fit2, adjust.method="fdr", p.value=p.cutoff, lfc=logfc.cutoff)
colnames(results) <- c('Adult', 'Pup', 'Embryonic')

GSE35802 <- gene_list
vennDiagram(results)
```

### Cross-reference embryonic effects against results from the Gu dataset re-analysis. 
Since GSE35802 was run on the Agilent platform we have to first collapse probes to unique genes. For each gene the probe with minimum p-value was retained. The Venn diagram illustrates the number of overlapping signigicant genes between the two datasets. The ROC curve is used to illustrate where the Gu dataset significant genes appear with respect to the ranking of GSE9202 genes (ranked by adjusted p-value).

```{r cross-reference1_option1, echo=FALSE, eval=FALSE, results='asis'}

# Pull out top genes and find the overlap
g.geo <- rep(0, nrow(GSE35802))
g.geo[which(rownames(GSE35802) %in% rownames(top.genes))] <- 1
g.gu <- rep(0, nrow(GSE35802))
g.gu[which(rownames(GSE35802) %in% rownames(gu.reanalysis))] <- 1

overlap <- (cbind(as.logical(g.geo), as.logical(g.gu)))
rownames(overlap) <- rownames(GSE35802)
colnames(overlap) <- c("GSE35802", "Gu Dataset")

vennDiagram(overlap)

# Use ROC to see where the Gu genes lie within the ranking
ordered <- GSE35802[order(GSE35802$adj.P.Val), ]
roc <- rocRanked(rownames(gu.reanalysis), ordered)

```

```{r cross-reference2_option2, echo=FALSE, eval=TRUE}

cnames <- c("GeneName", "GeneSymbol", "Entrez")
  
# Merge in the annotation data for GSE9202
columns <- c(10:12, 17, 18, 20, 21)
annot.geo1 <- GSE9202[, c(names(GSE9202)[columns])]
colnames(annot.geo1)[1:3]<-cnames

# Merge in the annotation data for GSE35802
columns <- c(8, 7, 6, 18, 19, 21, 22)
annot.geo2 <- GSE35802[, c(names(GSE35802)[columns])]
colnames(annot.geo2)[1:3]<-cnames

# Aggregate probes to unique genes using the minimum p-value
p2g.geo1 <- aggregate(. ~ GeneSymbol, data = annot.geo1[,c("GeneSymbol","adj.P.Val")], min)
p2g.geo2 <- aggregate(. ~ GeneSymbol, data = annot.geo2[,c("GeneSymbol","adj.P.Val")], min)
p2g.gu <- aggregate(. ~ Gene.Symbol, data = gu.reanlaysis[,c("Gene.Symbol","adj.P.Val")], min)

p2g.geo1$threshold.FDR = as.factor(p2g.geo1$adj.P.Val < p.cutoff)
p2g.geo2$threshold.FDR = as.factor(p2g.geo2$adj.P.Val < p.cutoff)

# Pull out top genes and find the overlap
g.gu <- rep(0, nrow(p2g.geo2))
g.gu[which(p2g.geo2$GeneSymbol %in% p2g.gu$Gene.Symbol)] <- 1

overlap <- (cbind(as.logical(p2g.geo2$threshold.FDR), as.logical(g.gu)))
rownames(overlap) <- p2g.geo2$GeneSymbol
colnames(overlap) <- c("GSE35802", "Gu Dataset")

vennDiagram(overlap)

# Use ROC to see where the Gu genes lie within the ranking
ordered <- data.frame(p2g.geo2[,2:3], row.names=p2g.geo2$GeneSymbol)
ordered <- ordered[order(ordered$adj.P.Val), ]
roc <- rocRanked(gu.reanalysis$Gene.Symbol, ordered)

```

### Intersecting all three datasets (at the gene level)

```{r intersection_option2, echo=FALSE}

g.geo1 <- rep(0, nrow(p2g.geo2))
sig <- p2g.geo1$GeneSymbol[which(p2g.geo1$threshold.FDR == "TRUE")]
g.geo1[which(p2g.geo2$GeneSymbol %in% sig)] <- 1


overlap <- (cbind(as.logical(p2g.geo2$threshold.FDR), as.logical(g.geo1), as.logical(g.gu)))
rownames(overlap) <- p2g.geo2$GeneSymbol
colnames(overlap) <- c("GSE35802", "GSE9202", "Gu Dataset")

vennDiagram(overlap)

```







